tests/build/build.go

```diff
@@ -396,14 +461,27 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
 
 		When("the component is removed and recreated (with the same name in the same namespace)", func() {
 			BeforeAll(func() {
-				Expect(f.AsKubeAdmin.HasController.DeleteHasComponent(componentName, testNamespace, false)).To(Succeed())
+				Expect(f.AsKubeAdmin.HasController.DeleteHasComponent(componentName, testNamespace, true)).To(Succeed())
 
+				timeout := 1 * time.Minute
+				interval := 1 * time.Second
 				Eventually(func() bool {
 					_, err := f.AsKubeAdmin.HasController.GetHasComponent(componentName, testNamespace)
 					return errors.IsNotFound(err)
-				}, time.Minute*1, time.Second*1).Should(BeTrue(), "timed out when waiting for the app %s to be deleted in %s namespace", applicationName, testNamespace)
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the app %s to be deleted in %s namespace", applicationName, testNamespace)
+				// Check removal of image repo
+				Eventually(func() bool {
+					imageExist, err := build.DoesImageRepoExistInQuay(imageRepoName)
+					return err == nil && !imageExist
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for image repo to be deleted")
+				// Check removal of robot account
+				Eventually(func() bool {
+					robotAccountExist, err := build.DoesRobotAccountExistInQuay(robotAccountName)
+					return err == nil && !robotAccountExist
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for robot account to be deleted")
```

Comment:
please update the inner functions passed to `Eventually` similarly as suggested in my other comment, thanks

Reply:
Done